src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -172,34 +156,91 @@ export class SearchInterceptor {
 
     const combinedSignal = getCombinedSignal(signals);
     const cleanup = () => {
-      this.timeoutSubscriptions.remove(subscription);
+      subscription.unsubscribe();
     };
 
     combinedSignal.addEventListener('abort', cleanup);
 
     return {
       combinedSignal,
+      timeoutSignal,
       cleanup,
     };
   }
 
-  // Right now we are debouncing but we will hook this up with background sessions to show only one
-  // error notification per session.
-  protected showTimeoutError = debounce(
-    (e: Error) => {
-      this.deps.toasts.addError(e, {
-        title: 'Timed out',
-        toastMessage: i18n.translate('data.search.upgradeLicense', {
-          defaultMessage:
-            'One or more queries timed out. With our free Basic tier, your queries never time out.',
+  /**
+   * Right now we are throttling but we will hook this up with background sessions to show only one
+   * error notification per session.
+   * @internal
+   */
+  private showTimeoutError = throttle((e: SearchTimeoutError) => {
```

Comment:
This was my take: https://github.com/elastic/kibana/pull/77788#discussion_r492054349

It seemed that debounce is not what we need here? Here is the case I thought of: 

debounce:

1. Search error:  ---> show popup right away
2. Search error in 5s ----> popup is debounced
3. in 60s debounce time is finished, since we had an error during the debounce ---> another popup will be shown


throttle: 

1. Search error:  ---> show popup right away
2. Search error in 5s ----> popup is ignored (throttled)
3. in 30s throttle is finished ---> no popup is shown

Reply:
OK,  played with it in sandbox: https://codesandbox.io/s/cool-breeze-fnlb7?file=/src/index.js

I think the right combination is: 

```
const showDebounceError = _.debounce(
  () => {
    console.log("debounced error...");
  },
  5000,
  { leading: true, trailing: false }
);
```

`trailing: false` avoids scenario I described above
and `debounce` delays next cycle for at least of `ms` between `showErrorCalls` 

@lizozom sorry for initial confusing with `throttle`.